RoslynPluginGenerator/NuGet/NuGetRepositoryFactory.cs

```diff
@@ -0,0 +1,68 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="NuGetRepositoryFactory.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+using NuGet;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SonarQube.Plugins.Roslyn
+{
+    /// <summary>
+    /// Creates the remote repository from which the packages will be fetched
+    /// </summary>
+    public static class NuGetRepositoryFactory
+    {
+        /// <summary>
+        /// Returns the settings loaded from the user- and machine-wide config files
+        /// </summary>
+        public static ISettings GetSettingsFromConfigFiles()
+        {
+            NuGetMachineWideSettings machineSettings = new NuGetMachineWideSettings();
+            ISettings settings = Settings.LoadDefaultSettings(null, null, machineSettings);
+            return settings;
+        }
+
+        /// <summary>
+        /// Creates and returns an aggregate repository using the specified settings
+        /// </summary>
+        public static IPackageRepository CreateRepository(ISettings settings, Common.ILogger logger)
```

Comment:
Some background on the relevant NuGet classes:
*  Settings: loaded from config files. Can specify multiple _package sources_ i.e. endpoints that can provide packages.
* PackageSourceProvider: class that can parse Settings and return the list of enabled packages sources.
* AggregateRepository: composite repository that wraps presents multiple separate package sources as a single repository.